packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -0,0 +1,855 @@
+///
+/// {@macro flutter.widgets.windowing.experimental}
+@internal
+abstract class RegularWindowController extends BaseWindowController {
+  /// Creates a [RegularWindowController] with the provided properties.
+  ///
+  /// Upon construction, the window is created by the platform.
+  ///
+  /// The [preferredSize] is the preferred content size of the window.
+  /// This might not be honored by the platform. This is the size that
+  /// the platform will try to apply to the window when it is created. In contrast,
+  /// the [preferredConstraints] field enforces the minimum and maximum size of
+  /// the window. If the [preferredSize] does not satisfy the [preferredConstraints]
+  /// or the [preferredSize] is null, then the platform will use an initial size
+  /// that does satisfy the [preferredConstraints] instead.
+  ///
+  /// The [preferredConstraints] are the constraints placed upon the size
+  /// of the window. This might not be honored by the platform.
+  /// This field enforces a minimum and maximum size on the window. If the
+  /// user attempts to resize the window beyond these constraints, the platform
+  /// will enforce the constraints according to its own policy. For example, the
+  /// platform might clip the content to fit within the resized window, or it might
+  /// prevent the window from being resized altogether. If null, the window wil
+  /// be unconstrained.
+  ///
+  /// If the [preferredSize] is null, then the platform will use an
+  /// initial size that satisfies the [preferredConstraints].
+  ///
+  /// If the [preferredSize] is not null and it does  not satisfy the
+  /// [preferredConstraints], then the platform will use an
+  /// initial size that does satisfy the [preferredConstraints] instead.
+  ///
+  /// If both [preferredSize] and [preferredConstraints] are null,
+  /// then the platform will use its own default size for the window.
+  ///
+  /// The [title] argument configures the window's initial title.
+  /// If omitted, some platforms might fall back to the app's name.
+  ///
+  /// The [delegate] argument can be used to listen to the window's
+  /// lifecycle. For example, it can be used to save state before
+  /// a window is closed.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  factory RegularWindowController({
+    Size? preferredSize,
+    BoxConstraints? preferredConstraints,
+    String? title,
+    RegularWindowControllerDelegate? delegate,
+  }) {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_kWindowingDisabledErrorMessage);
+    }
+
+    final WindowingOwner owner = WidgetsBinding.instance.windowingOwner;
+    final RegularWindowController controller = owner.createRegularWindowController(
+      delegate: delegate ?? RegularWindowControllerDelegate(),
+      preferredSize: preferredSize,
+      preferredConstraints: preferredConstraints,
+    );
+    if (title != null) {
+      controller.setTitle(title);
+    }
```

Comment:
Would it make sense to simplify this by adding a `title` argument to `WindowingOwner. createRegularWindowController`? I don't feel strongly about this, feel free to skip.